### PR TITLE
fix: Rest endpoint providing item class list

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
Allow endpoint to provide list of item classes with simple query and page control.

New endpoint:
`taoItems/RestItems/getItemClasses?q=item&page=1`

### CODE REVIEW

**Comment by tikhanovichA** [comment]
Could you pleaase fix those issues in scope of this PR? 
image

**Review by tikhanovichA — APPROVED** [review]
**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code respects best practices
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

File: models/classes/search/ItemClassListServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+namespace oat\taoItems\model\search;
+
+use oat\generis\model\data\Ontology;
+use oat\generis\model\DependencyInjection\ContainerServiceProviderInterface;
+use oat\generis\model\kernel\persistence\smoothsql\search\ComplexSearchService;
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+
+use function Symfony\Component\DependencyInjection\Loader\Configurator\service;
+
+class ItemClassListServiceProvider implements ContainerServiceProviderInterface
```
Comment (Karol-Stelmaczonek) [inline models/classes/search/ItemClassListServiceProvider.php:32]: We don't need to register it in manifest?

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**